### PR TITLE
DiskRP swagger 2021-12-01

### PR DESCRIPTION
**Pull request by rytruong**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [x] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  4. If updating an existing version, please select the specific langauge SDKs and CLIs that must be refreshed after the swagger is published.
      - [x] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [x] PowerShell
      - [x] CLI
      - [ ] Terraform
      - [ ] No refresh required for updates in this PR

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist 

> **Applicability**: :warning: 
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs 
> - Adding new properties 
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [x] Check this box if any of the following apply to the PR so that label "WaitForARMFeedback" will be added automatically to begin ARM API Review. Failure to comply may result in delays to the manifest.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you are using OpenAPIHub to initialize the PR for adding a new version. More details, refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).

- [x] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If any of the following scenarios apply to the PR, request approval from the Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 
- [ ] Removing API(s) in a stable version
- [ ] Removing properties in a stable version
- [ ] Removing API version(s) in a stable version
- [ ] Updating API in a stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/compute/resource-manager/Microsoft.Compute/stable/2021-08-01/disk.json [hunk 1]
```diff
@@ -3,7 +3,7 @@
   "info": {
     "title": "DiskResourceProviderClient",
     "description": "The Disk Resource Provider Client.",
-    "version": "2021-08-01"
+    "version": "2021-12-01"
```
[the remark this answers is not in the document]
Comment (rytruong) [reply specification/compute/resource-manager/Microsoft.Compute/stable/2021-08-01/disk.json]: updated it

File: specification/compute/resource-manager/Microsoft.Compute/stable/2021-12-01/disk.json [hunk 2]
```diff
@@ -2590,6 +2602,28 @@
         ]
       }
     },
+    "DataAccessAuthMode": {
+      "type": "string",
+      "description": "Add additional AAD authentication requirement when accessing disk/snapshot export URL.",
```
Comment (chetmaddula) [inline specification/compute/resource-manager/Microsoft.Compute/stable/2021-12-01/disk.json:2607]: when accessing export/upload URL

Comment (rytruong) [reply specification/compute/resource-manager/Microsoft.Compute/stable/2021-12-01/disk.json]: updated

File: specification/compute/resource-manager/Microsoft.Compute/stable/2021-12-01/disk.json [hunk 3]
```diff
@@ -2590,6 +2602,28 @@
...
+        "AzureActiveDirectory",
+        "None"
+      ],
+      "x-ms-enum": {
+        "name": "DataAccessAuthMode",
+        "modelAsString": true,
+        "values": [
+          {
+            "value": "AzureActiveDirectory",
+            "description": "When export/upload URL is used additional authentication is performed, to check if the user has necessary RBAC access. Please refer to aka.ms/{placeholder}."
```
Comment (chetmaddula) [inline specification/compute/resource-manager/Microsoft.Compute/stable/2021-12-01/disk.json:2618]: When export/upload URL is used, the system checks if the user has an identity in Azure Active Directory and has necessary permissions to export/upload the data.

Comment (chetmaddula) [reply specification/compute/resource-manager/Microsoft.Compute/stable/2021-12-01/disk.json:2618]: Please refer to [aka.ms/DisksAzureADAuth](https://aka.ms/DisksAzureADAuth)

File: specification/compute/resource-manager/Microsoft.Compute/stable/2021-12-01/disk.json [hunk 4]
```diff
@@ -2590,6 +2602,28 @@
         ]
       }
     },
+    "DataAccessAuthMode": {
+      "type": "string",
+      "description": "Add additional AAD authentication requirement when accessing export/upload URL.",
```
Comment (vikramd-ms) [inline specification/compute/resource-manager/Microsoft.Compute/stable/2021-12-01/disk.json:2607]: Additional authentication requirements when exporting or uploading to a disk or snapshot

File: specification/compute/resource-manager/Microsoft.Compute/stable/2021-12-01/disk.json [hunk 5]
```diff
@@ -2954,9 +2994,21 @@
         "acceleratedNetwork": {
           "type": "boolean",
           "description": "True if the image from which the OS disk is created supports accelerated networking."
+        },
+        "architecture": {
+          "type": "string",
+          "description": "The architecture of the image.",
```
Comment (vikramd-ms) [inline specification/compute/resource-manager/Microsoft.Compute/stable/2021-12-01/disk.json:3000]: CPU architecture supported by an OS disk

File: specification/compute/resource-manager/Microsoft.Compute/stable/2021-12-01/disk.json (hunk 3 above)
Comment (rytruong) [reply specification/compute/resource-manager/Microsoft.Compute/stable/2021-12-01/disk.json]: updated

File: specification/compute/resource-manager/Microsoft.Compute/stable/2021-12-01/disk.json (hunk 4 above)
Comment (rytruong) [reply specification/compute/resource-manager/Microsoft.Compute/stable/2021-12-01/disk.json]: updated

File: specification/compute/resource-manager/Microsoft.Compute/stable/2021-12-01/disk.json (hunk 5 above)
Comment (rytruong) [reply specification/compute/resource-manager/Microsoft.Compute/stable/2021-12-01/disk.json]: updated

**Comment by haagha** [comment]
@ArcturusZhang can you help with the Track2 SDK generation failures and breaking change checks?

**Comment by ArcturusZhang** [comment]
Hi @rytruong I was informed that this PR is under some internal communications therefore the merge need to be hold for a few time - I added a "DoNotMerge" label on this PR. Please feel free to ping me when this is ready to merge

**Comment by grizzlytheodore** [comment]
@ArcturusZhang  this is ready for merge

**Review by chetmaddula — APPROVED** [review]

**Comment by grizzlytheodore** [comment]
last commit is syncing with change that went into 2021-08-01 version few days ago https://github.com/Azure/azure-rest-api-specs/pull/17633

**Comment by ArcturusZhang** [comment]
Well. the x-ms-identifiers is causing some issues for the track 1 generation. I am fixing it in [this PR](https://github.com/Azure/azure-rest-api-specs/pull/17853) and merge that first.

**Review by ArcturusZhang — APPROVED** [review]

**Pull request merged**